### PR TITLE
Added Root container for routes and App for homepage

### PR DESCRIPTION
**Pull request by mallett002**
Root is wrapped in Provider to have access to Redux Store. Root also renders the routes. App is a Redux container component. Still need to make the reducers and make the state shape. Then I can make the presentational React components.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,8 +1,14 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import './index.css';
-import App from './App';
-import registerServiceWorker from './registerServiceWorker';
+import React from "react"
+import { render } from "react-dom"
+import { BrowserRouter as Router } from 'react-router-dom'
+import Root from './containers/Root'
+import store from './store/configureStore'
```
Comment (mmallett) [inline src/index.js]: `;` at end of `import` statements

File: src/containers/App.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        errorMessage: PropTypes.string,
+        // Some more props will be added...
+    }
+
+    renderErrorMessage() {
+        const { errorMessage } = this.props
+        if (!errorMessage) {
+            return null
+        }
+        // Will possibly change this to something more usefull in the future...
```
Comment (mmallett) [inline src/containers/App.js]: useful

Comment (mmallett) [reply src/containers/App.js]: Add TODO to the front of the comment. Anytime you want to return to something later, add a TODO comment

File: src/containers/App.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { withRouter } from 'react-router-dom'
+import LinkedInButton from '../components/LinkedInButton'
+import { fetchLinkedIn } from '../actions/actions'
```
[the remark this answers is not in the document]
Comment (mallett002) [reply src/containers/App.js]: Ok. Most of the official docs left it out, so I did. But i used to add those in. I'll put em in :)

File: src/containers/App.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+class App extends Component {
+    static propTypes = {
+        fetchLinkedIn: PropTypes.func.isRequired,
+        errorMessage: PropTypes.string,
+        // Some more props will be added...
```
[the remark this answers is not in the document]
Comment (mallett002) [reply src/containers/App.js]: Thought it was how you check PropTypes in class component. Just read that you need to import "transform-class-properties" to use it like that. Normally it goes outside the class like this:
App.propTypes = {
  fetchLinkedIn: PropTypes.func.isRequired,
  // TODO... 
}

I'll put them in like that.

**Pull request merged**